Content.Server/RevolutionFlag/FlagSystem.cs

```diff
@@ -0,0 +1,96 @@
+using Content.Server.RevolutionFlag.Components;
+using Content.Server.Mind.Components;
+using Content.Server.Traitor;
+using Content.Server.Weapon.Melee;
+using Content.Shared.Damage;
+using Content.Shared.Hands;
+using Content.Shared.StatusEffect;
+
+namespace Content.Server.RevolutionFlag;
+    internal sealed class FlagSystem : EntitySystem
+    {
+        [Dependency] private readonly EntityLookupSystem _entityLookup = default!;
+        [Dependency] private readonly StatusEffectsSystem _statusEffectSystem = default!;
+        private const string RevolutionaryPrototypeId = "Revolutionary";
+        private const string CourageEffectKey = "Courage";
+        private DamageModifierSet Modifiers = default!; 
+        
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<FlagBuffComponent,MeleeHitEvent>(ModifyDamage);
+            SubscribeLocalEvent<FlagComponent,GotEquippedHandEvent>(OnEquipped);
+            SubscribeLocalEvent<FlagComponent,GotUnequippedHandEvent>(OnDequipped);
+        }
+
+        private void OnEquipped(EntityUid uid, FlagComponent comp, GotEquippedHandEvent args)
+        {
+            comp.active = true;
+        }
+
+        private void OnDequipped(EntityUid uid, FlagComponent comp, GotUnequippedHandEvent args)
+        {
+            comp.active = false;
+        }
+
+        public void Aura(FlagComponent flagComp, float range)
+        {
+            var transform = EntityManager.GetComponent<TransformComponent>(flagComp.Owner);
+            var mapPosition = transform.MapPosition;
+            var inRange = _entityLookup.GetEntitiesInRange(transform.Coordinates, range);
+
+            if (!flagComp.active)
+                return;
+            foreach(var entityInRange in inRange)
+            {
+                if (!TryComp<MindComponent>(entityInRange, out var mind))
+                    continue;
+                if (!GetRole(mind, RevolutionaryPrototypeId))
+                    continue;
+                _statusEffectSystem.TryAddStatusEffect<FlagBuffComponent>(entityInRange, CourageEffectKey, flagComp.timespan, true);
+            }
+        }
+
```

Comment:
Don't use entitylookup to query these, use collision as this is going to be too expensive. You also don't seem to remove the statuseffect?